From b4952044f8de9be6d21b8e0da328d877a612689c Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Dec 2024 21:04:08 +0800
Subject: [PATCH 172/273] AOSCOS: MIPS: Kconfig: mark LOONGSON64_CPUAUTOPLUG as
 bool instead

The symbols for functions lock_device_hotplug() and
unlock_device_hotplug() are not exported, preventing this driver from
being built as a module.

Disallow building as module by changing the driver's configuration from
tristate to bool.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 1ef06048dc61..8ec6becf599c 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -1683,7 +1683,7 @@ config CPU_BMIPS
 endchoice
 
 config LOONGSON64_CPUAUTOPLUG
-        tristate "Loongson-3A CPU Autoplug Driver"
+        bool "Loongson-3A CPU Autoplug Driver"
         depends on CPU_LOONGSON64 && HOTPLUG_CPU
         help
 	  This option adds a CPU autoplug driver for Loongson-3 multi-core
-- 
2.47.1

